### PR TITLE
Move regression test for #14291 into compatibility tests

### PR DESCRIPTION
**Pull request by akrmn**
By doing this, we
1. remove one `.dar` from version control
2. test data-dependencies built with each stable SDK version (>=1.18.0) (instead of just `2.2.0`)

Additionally, the new definition of `daml_script_test` makes it easier to write daml-script tests under `compatibility/`.

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Please push a commit (possibly empty) with the

```
run-full-compat: true
```

trailer.

**Comment by garyverhaegen-da** [comment]
```
sdk-version: 1.0.1
[...]
Message: 
  src/Dep.daml:25:18: error:
  • Variable not in scope: archiveCmd : ContractId T0 -> Commands b0
  • Perhaps you meant ‘archive’ (imported from Prelude)
```

It looks like [archiveCmd was added in 1.5](https://github.com/digital-asset/daml/commit/6743bf5b9becda9f5632ad7e409d6b2b66d2be3d), so you'll have to add bounds on this test so it doesn't run on versions before that.

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM. Thanks.

File: compatibility/bazel_tools/data_dependencies/data_dependencies.bzl [hunk 1]
```diff
@@ -106,16 +121,64 @@ def data_dependencies_upgrade_test(old_sdk_version, new_sdk_version):
...
-            "$(rootpath %s)" % dar_name,
+        dar_name = dar_name,
+        sdk_version = new_sdk_version,
+    )
+
+# regression test for https://github.com/digital-asset/daml/issues/14291
+def data_dependencies_daml_script_test(old_sdk_version):
+    data_dep_name = "data-dependencies-script1-{old_sdk_version}".format(
+        old_sdk_version = old_sdk_version,
+    )
```
[the remark this answers is not in the document]
Comment (akrmn) [reply compatibility/bazel_tools/data_dependencies/data_dependencies.bzl:137]: tbh I'd rather keep the explicit version

**Pull request merged**